src/System.Private.CoreLib/shared/System/Buffers/Text/FormattingHelpers.CountDigits.cs

```diff
@@ -128,5 +128,34 @@ public static int CountHexDigits(ulong value)
 
             return digits;
         }
+
+        
+        // Counts the number of trailing '0' digits in a decimal numnber.
+        // e.g., value =      0 => retVal = 0, valueWithoutTrailingZeros = 0
+        //       value =   1234 => retVal = 0, valueWithoutTrailingZeros = 1234
+        //       value = 320900 => retVal = 2, valueWithoutTrailingZeros = 3209
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int CountDecimalTrailingZeros(uint value, out uint valueWithoutTrailingZeros)
+        {
+            int zeroCount = 0;
+
+            if (value != 0)
+            {
+                while (true)
```

Comment:
Couldn't the `if` statement be removed and the loop invariant changed to `while (value != 0)`?

Reply:
Yes, but it would add an extra comparison every iteration of the loop after the first.